### ISSUE TITLE
setuptools scm workaround doesnt work for packages with dashes in the name

### ISSUE THREAD

**Issue by RonnyPfannschmidt** [mgedmin/check-manifest#145]
the fix of #68 doesn't notify the user that it uses pretend_version,, and incorrectly puts in a bad version number

this triggers https://github.com/pypa/setuptools_scm/issues/630 

as https://github.com/zobayer1/logging-extras has in fact a extra dash

its ABSOLUTELY necessary to inform the users if something like `SETUPTOOLS_SCM_PRETEND_VERSION` is set
additionally its not safe to use for build processes that actually install dependencies (as any pulled sdist that wants setuptools_scm may get misinformed about their versions

https://github.com/mgedmin/check-manifest/blob/7c4403a845b5d7bf3ad1ba930daf192c049182cc/check_manifest.py#L877-L882 incorrctly extract

and then https://github.com/mgedmin/check-manifest/blob/7c4403a845b5d7bf3ad1ba930daf192c049182cc/check_manifest.py#L977 always  replaces without a warning (so people cant even fix it with a env var)

**Comment by zobayer1** [comment]
I faced [pypa/setuptools_scm#630](https://github.com/pypa/setuptools_scm/issues/630) only recently. However,  @RonnyPfannschmidt  made a quick investigation that led me here. I am not sure whether a dash should or should not be part of a source distribution name, but it seems to be a fairly common practice to use dashes in the name. In fact, the same is done in `check-manifest` itself.
https://github.com/mgedmin/check-manifest/blob/7c4403a845b5d7bf3ad1ba930daf192c049182cc/setup.py#L29

As pointed out above, the code here assumes that there would not be any dashes in the file name.
https://github.com/mgedmin/check-manifest/blob/7c4403a845b5d7bf3ad1ba930daf192c049182cc/check_manifest.py#L882

Knowing that setuptools version strings do not contain extra dashes, this could be replaced with the following

```py
    return filename.split('-')[-1]
```

I have tested a build containing this change with my setup, and the tests provided with this repo. It is working as expected now.

Was there a specific reason or motivation to use the `partition` method there? Were there any special cases that `split` would not be able to handle?
